server/manager/src/main/java/org/apache/accumulo/master/tableOps/create/ChooseDir.java

```diff
@@ -60,9 +63,16 @@ public long isReady(long tid, Master environment) {
 
   @Override
   public void undo(long tid, Master master) throws Exception {
-    Path p = tableInfo.getSplitDirsPath();
-    FileSystem fs = p.getFileSystem(master.getContext().getHadoopConf());
-    fs.delete(p, true);
+    // Clean up split files if ChooseDir operation fails
+    try {
+      if (tableInfo.getInitialSplitSize() > 0) {
+        Path p = tableInfo.getSplitDirsPath();
+        FileSystem fs = p.getFileSystem(master.getContext().getHadoopConf());
+        fs.delete(p, true);
+      }
+    } catch (NullPointerException | IOException e) {
+      log.error("Failed to undo ChooseDir operation", e);
```

Comment:
When on the receiving end of an exception like this its really nice to have some information that help correlates it with other activity in the system.  However the danger of trying to get more information for an exception is that code made fail.  I think the following is pretty safe and gives a good bit of helpful info. 

```suggestion
      var spdir = Optional.ofNullable(tableInfo).map(TableInfo::getSplitDirsPath).orElse(null);
      log.error("{} Failed to undo ChooseDir operation, split dir {} ",FateTxId.format(tid), spdir, e);
```

Reply:
I made the requested error changes or each of the catch blocks. Let me know if there is anything else I need to tweak.

Reply:
Looks good.